src/debugger.ts

```diff
@@ -2,32 +2,70 @@ import * as vscode from 'vscode';
 import * as path from 'node:path';
 import { globalContext } from './extension';
 import { MirrordConfigManager } from './config';
-import { LAST_TARGET_KEY, MirrordAPI, mirrordFailure } from './api';
+import { LAST_TARGET_KEY, MirrordAPI, mirrordFailure, MirrordExecution } from './api';
 import { updateTelemetries } from './versionCheck';
 import { getLocalMirrordBinary, getMirrordBinary } from './binaryManager';
+import { platform } from 'node:os';
 
-/// Get the name of the field that holds the exectuable in a debug configuration of the given type.
-function getExecutableFieldName(config: vscode.DebugConfiguration): keyof vscode.DebugConfiguration {
+const DYLD_ENV_VAR_NAME = "DYLD_INSERT_LIBRARIES";
+
+/// Get the name of the field that holds the exectuable in a debug configuration of the given type,
+/// and the executable. Returning the field name for replacing the value with the patched path later.
+/// Also returning the executable because in some configuration types there is some extra logic to
+/// be done for retrieving the executable out of its field (see the `node-terminal` case).
+function getFieldAndExecutable(config: vscode.DebugConfiguration): [keyof vscode.DebugConfiguration, string | null] {
 	switch (config.type) {
 		case "pwa-node":
 		case "node": {
-			return "runtimeExecutable";
+			return ["runtimeExecutable", config["runtimeExecutable"]];
+		}
+		case "node-terminal": {
+			return ["command", config["command"]?.split(' ')[0]];
 		}
 		case "python": {
 			if ("python" in config) {
-				return "python";
+				return ["python", config["python"]];
 			}
 			// Official documentation states the relevant field name is "python" (https://code.visualstudio.com/docs/python/debugging#_python), 
 			// but when debugging we see the field is called "pythonPath".
-			return "pythonPath";
+			return ["pythonPath", config["python"]];
 		}
 		default: {
-			return "program";
+			return ["program", config["program"]];
 		}
```

Comment:
wait, do we handle the python case as well? not clear, what if the command is `python3 foo bar`?

Reply:
We change the `command` attribute in configurations with configuration type `node-terminal`.
If the command is `python3 foo bar` debugging wouldn't work, because vscode would not be able to connect the node debugger.
If there is a configuration type where you can set `"command": "python3 foo bar"`, we don't support SIP sidestepping for it yet, and we need to add support for it in a separate PR.

Reply:
I was referring to python configuration, would it fail just like the case of `node-terminal`?

Reply:
No, python configurations (of type `python`) would not fail if they have the configuration fields we know about (`python`/`pythonPath`), as they are already supported, it's already there.

The `node-terminal` configuration type was not supported here up until now and this is why it would fail. We are adding support for it now. We already have a `python` case in this `switch`, where we handle the case where the configuration type is `python`.

If, like `node-terminal` for Node.js, there is a configuration type for Python we're not aware of, it would fail. We need to add SIP-sidestepping support for each configuration type we want to support, and this PR adds it for `node-terminal`. Please also note the explanation in the [PR's text](https://github.com/metalbear-co/mirrord-vscode/pull/27#issue-1803440246).

Reply:
cool, makes sense